src/xbot/common/subsystems/ConnectionMonitorSubsystem.java

```diff
@@ -0,0 +1,46 @@
+package xbot.common.subsystems;
+
+import com.google.inject.Singleton;
+import edu.wpi.first.wpilibj.Timer;
+import xbot.common.command.BaseSubsystem;
+import xbot.common.logic.Latch;
+import xbot.common.properties.DoubleProperty;
+import xbot.common.properties.XPropertyManager;
+
+import java.util.Observable;
+import java.util.Observer;
+
+@Singleton
+public class ConnectionMonitorSubsystem extends BaseSubsystem implements Observer {
+
+    protected final DoubleProperty timeOut;
+    private final Latch connectionLatch = new Latch(true, Latch.EdgeType.Both);
```

Comment:
@JohnGilb @WasabiFan when I choose the Falling edge type here it will only work once when I try to reset the state by forcing a rising edge on the test. It won't detect the next falling edge.

Reply:
Does the Latch test class test for that? If not, you should add a test. If it fails, you can fix the Latch. If it passes, then you need to figure out what you're doing differently.

Reply:
Fixed